[ghos7ie] iP

ghos7ie opened the pull request:
# LBot

> Your personal dedicated hyper assistant 

LBot has the following functionalities:
- text-based
- easy to learn
- if it doesn't work? It's called LBot for a reason ;)

All you need to do is:
1. Download the **latest jar** from [here](https://github.com/ghos7ie/ip/releases).
2. Run it.
3. Add your tasks.
4. And LBot will manage the rest for you :)

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon?)
- [ ] Reminders (~coming probably never😉~)

---
If you're a Java programmer, here's the ```main``` method for you!
You can also update the path where you wish to save your tasks!
```java
    /**
     * Main class of LBot.
     */
    public static void main(String[] args) throws LBotException {
        new LBot("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ghos7ie committed:
Completed Level 0: Rename, Greet, Exit

ghos7ie committed:
Completed Level 0: Rename, Greet, Exit

ghos7ie committed:
Level 1 Echo

ghos7ie committed:
Level 1 Update

Added scanner.close() and restructured loop

ghos7ie committed:
Level 2: Add and List

ghos7ie committed:
Level 3: Mark as Done

ghos7ie committed:
Add Task, Todo, Deadline and Event classes

ghos7ie committed:
Add Delete functionality

ghos7ie committed:
Add TextUiTesting

ghos7ie committed:
Update Exceptions

Unable to figure out how to use custom exception classes

ghos7ie committed:
Add Status Enum

Allows for future expansion of task status rather than keeping it as a boolean

ghos7ie committed:
Add Exception classes, Add PrintFormatter class

ghos7ie committed:
Update PrintFormatter.java

ghos7ie committed:
Checkpoint 1

ghos7ie committed:
Add additional OOP classes

ghos7ie committed:
Format dates

ghos7ie committed:
Update .gitignore

ghos7ie committed:
Delete tasks.txt

ghos7ie committed:
Merge branch 'master' into branch-Level-8

ghos7ie committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Format dates

ghos7ie committed:
Add Save and Load function

ghos7ie committed:
Merge branch 'branch-Level-7'

ghos7ie committed:
Add Package

ghos7ie committed:
Add packages

ghos7ie committed:
Add reply messages

ghos7ie committed:
Merge branch 'add-gradle-support'

ghos7ie committed:
Add Gradle and JUnit Tests

ghos7ie committed:
Add JavaDoc comments

ghos7ie committed:
Merge branch 'A-JavaDoc'

ghos7ie committed:
Implement Find

ghos7ie committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Implement Find

ghos7ie committed:
Comply with Checkstyle

ghos7ie committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Comply with Checkstyle

ghos7ie committed:
Complete Level 10

ghos7ie committed:
Merge branch 'branch-Level-10'

ghos7ie committed:
Comply with CheckStyle

ghos7ie committed:
Update README.md

kohkakohla commented on src/main/java/lbot/task/Event.java:
```diff
@@ -0,0 +1,52 @@
+    public String toString() {
+        return this.type + "[" + this.status() + "] " + this.description
+                + " (from: " + this.startDateTime.format(dateTimeFormat)
+                + " to: " + this.endDateTime.format(dateTimeFormat) + ")";
```
I like how you formatted this long line into small lines!

kohkakohla reviewed (approved):
LGTM! Really well done code that follows coding standards very well. Very easy to read code as well.

trpmy commented on src/main/java/lbot/command/Command.java:
I like how you used OOP in abstracting the commands as its own class to avoid using switch case!

trpmy commented on src/main/java/lbot/Launcher.java:
```diff
@@ -0,0 +1,18 @@
+import lbot.gui.Main;
+
+/**
+ * LStart point of LBot.
```
Super minor nit here, but might this be a typo in your header comment or a style choice in saying `LStart point`, as in the same branding as `LBot` perhaps?

ghos7ie replied:
hey thanks! didn't catch that mistake, will fix it soon :)

trpmy commented on config/checkstyle/checkstyle.xml:
I like how you set up the checkstyle script to make adhering to coding style guide easier!

trpmy commented:
Overall, I find your code quality to be quite excellent! I like how used a lot of OOP principles to great advantage, both in making your code highly readable and ensuring maintainability.

ghos7ie committed:
Fix typo

ghos7ie committed:
Add assertion with correct usage

Incorrect usage of assertions in MainWindow.java.
No checks were done in the code prior to where it was used.

Fix is to use assertion at the appropriate location.

Added correct usage in LBot.java class since method used should
throw exception if an incorrect command was supplied.

ghos7ie committed:
Merge pull request #1 from ghos7ie/branch-A-Assertions

Add assertion with correct usage

ghos7ie committed:
Add 'f' command for find

Allow user to use both 'f' and 'find' to execute find command.

Uses stream to allow for no side effects on the main TaskList.

Takes advantage of filter() function of Java streams to allow for
lesser code to be written.

ghos7ie committed:
Merge branch 'master' into branch-A-Streams

ghos7ie committed:
Merge pull request #2 from ghos7ie/branch-A-Streams

Branch a streams

ghos7ie committed:
Refractor unused codes and methods

ghos7ie committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

ghos7ie committed:
Merge pull request #3 from ghos7ie/branch-A-CodeQuality

Branch a code quality

ghos7ie committed:
Create gradle.yml

ghos7ie committed:
Remove extra workflows

ghos7ie committed:
Fix checkstyle

ghos7ie committed:
Fix Junit test for event

ghos7ie committed:
Update Bye Command

ghos7ie committed:
Remove unused import and change javadoc comment

ghos7ie committed:
Create UG

ghos7ie committed:
Update README.md

ghos7ie committed:
Update error handling

ghos7ie committed:
fix regex
